oeps/oep-0007.rst

```diff
@@ -0,0 +1,462 @@
+some behaviors that differ by default between Python 2 and 3. `from
+__future__ import absolute_import` forbid the use of implicit relative
+imports and `from __future__ import print_function` to make print a function
+instead of a statement.  `from __future__ import division` will make
+single-slash division (`a / b`) always perform floating point division, and
+double-slash division (`a // b`) always truncate down.
+
+Text handling
+-------------
+
+Text handling is the largest area of difficulty in porting Python.  Where
+possible, we will use unambiguous text or byte objects.  In most cases, text
+should be preferred.  Bytes should only be used when you can answer the
+question: "Do I need this specific sequence of bytes."  The most
+error-resistent way to acheive this is to use what is called a "unicode
+sandwich."  This means that as soon as you receive data from a file or network
+interface, it should be converted to text. Your code should then treat it as
+text for as long as possible, only encoding it back to bytes when sending it to
+an interface that requires bytes (such as a file, a network interface, or a
+bytes-oriented library). The only operation that should (ideally) be performed
+on bytes is decoding.
+
+In those cases where ambiguity is required (such as working with libraries like
+`csv` which require byte strings in Python 2 and unicode strings in Python 3,
+we should isolate the need for ambiguity as much as possible.  Type checking
+libraries like `PyContracts` (already used in edx-platform) or `typing` (a
+backport of the type hinting system introduced in Python 3.5) can help us
+ensure that callers are using the appropriate variety of string.
+
+If you need to create bytes, and there is no compelling reason to use a
+specific encoding, use utf-8.  Compelling reasons include requirements of a
+particular data format or protocol, or requirements of legacy or third-party
+libraries.
+
+If you need to accept bytes, and we have the freedom to require a particular
+encoding, require utf-8.  If we need to support multiple encodings, require
+that inputs specify their encoding explicitly, or be treated as utf-8.  Refuse
+the temptation to guess anything other than utf-8.  Misencoded inputs should
+ideally be rejected as an error. If that is not an option, malformed characters
+should be replaced with the unicode replacement character,
+[U+FFFD](http://unicode-table.com/en/FFFD/).  If you need to accept bytes from
+an interface that doesn't specify its encoding, pass it through a wrapper that
+does specify the encoding, and use that wrapper instead.
+
+There are two major ways of handling text and byte literals uniformly across
+Python versions.  We do not explicitly require one way over the other, but
+decisions should be made on a per-project basis, and adhered to by all
+developers working on that project.
+
+Handling literals, Option 1: Python 3-Style
+...........................................
```

Comment:
Periods don't seem to be among the [valid section header underline characters](http://docutils.sourceforge.net/docs/user/rst/quickref.html#section-structure)